### PR TITLE
[docs] Add documentation on @NullablePosition

### PR DESCRIPTION
**Pull request by ZacBlanco**
## Description

Adding docs for the `@NullablePosition` annotation

## Motivation and Context

Missing docs.

## Impact

N/A

## Test Plan

N/A

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-docs/src/main/sphinx/develop/functions.rst [hunk 1]
```diff
@@ -336,15 +336,18 @@ the standard Java types can't be used as they aren't able to represent the input
 data.
 
 To define a method handle which can accept *any* types, use ``@BlockPosition``
-in conjunction with the ``@BlockIndex`` parameters. This works for both scalar
-and aggregation function implementations. An example signature is provided
-below. 
+in conjunction with the ``@BlockIndex`` parameters. Similar to the
+``@SqlNullable`` annotation, use the ``@NullablePosition`` annotation to denote
+that the function should be called when the block position is null.
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/functions.rst:341]:
```suggestion
that the function should be called when the block position is ``NULL``.
```

Formatting suggested for consistency with lines 59, 71-76, 176.

File: presto-docs/src/main/sphinx/develop/functions.rst [hunk 2]
```diff
@@ -336,15 +336,18 @@ the standard Java types can't be used as they aren't able to represent the input
...
+
+This works for both scalar and aggregation function implementations. An example
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/develop/functions.rst:343]:
```suggestion
This works for both scalar and aggregation function implementations. 
```
Suggest deleting the sentence "An example signature is provided below." as the other examples in this section aren't introduced. If I'm overlooking something meaningful here, just let me know and then ignore this suggestion.

**Review by steveburnett — CHANGES_REQUESTED** [review]
One nit, and one suggestion that you're welcome to ignore. Thanks!

File: presto-docs/src/main/sphinx/develop/functions.rst (hunk 1 above)
Comment (ZacBlanco) [reply presto-docs/src/main/sphinx/develop/functions.rst]: fixed

File: presto-docs/src/main/sphinx/develop/functions.rst (hunk 2 above)
Comment (ZacBlanco) [reply presto-docs/src/main/sphinx/develop/functions.rst]: fixed

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

Thanks!

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**